### ISSUE TITLE
django channel save chat record upon last user leave the room

### ISSUE THREAD

**Issue by gotexis** [django/channels#1188]
Hi I am new to any Async coding hence the great django-channels official lib is a bit obfuscated to me.

I was able to define a ORM process upon message send to save the message to db. That will be easy.

However, imagine 5000 users sending messages simultaneously. The db will easily be overloaded even if I use a celery task queue. Plus the db could easily accumulate millions of rows in a short period.

I imagine there to be a way to gather all lingering messages in the channel-layer, and upon layer close (last user exit, websocket connection = 0), save these into a JsonField as one row?

**TLDR: how to check layer termination, and after that how to get all messages sent? Where are they stored, in memory?**

Current thought on a naive method: create a memory register (like database, but in memory only) for each channel and place all messages there as they sent. Then when the websocket terminates, trigger a save code. But then, **how to create this kind of fake-db that only exist in memory?**

Appreciate all suggestions! Please point out if I am going the wrong direction in any way.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by andrewgodwin** [comment]
Sorry, but we can't help out with general support requests here - the issue tracker is for reproduceable bugs and concrete feature requests only! Please see our support documentation (http://channels.readthedocs.io/en/latest/support.html)
for more information about where you can get general help.
